### PR TITLE
Arm/master/update rcar 2.0.0

### PR DESCRIPTION
**Pull request by marex**
This patchset updates the R-Car Gen3 ATF support, which was based on Renesas ATF release v1.0.22, with select changes from release v2.0.0. The patchset contains various cleanups, fixes, and updates to QoS/PFC/DDR initialization. There are still a couple of odd changes in the downstream ATF which need further inspection which are not part of this patchset and will be submitted separately later if applicable.

Tested on H3 ES2.0 ULCB, Salvator-XS ; M3W ES1.0 ULCB, Salvator-X, M3N ES1.0 Salvator-XS, E3 ES1.0 Ebisu-4D .

### CODE REVIEW

**Comment by antonio-nino-diaz-arm** [comment]
Looks good to me. IP Review +1

**Comment by antonio-nino-diaz-arm** [comment]
You need to rebase this to fix the conflicts.

**Comment by marex** [comment]
On 1/4/19 4:11 PM, Antonio Niño Díaz wrote:
> You need to rebase this to fix the conflicts.

What conflicts ? I verified it is on tip of master branch,
c8765826f4c2d10db0b660defccc84f7bce11af0 , before I submitted this.

-- 
Best regards,
Marek Vasut

File: plat/renesas/rcar/platform.mk [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # SPDX-License-Identifier: BSD-3-Clause
 #
 
-PROGRAMMABLE_RESET_ADDRESS	:= 0
+PROGRAMMABLE_RESET_ADDRESS	:= 1
```
Comment (ldts) [inline plat/renesas/rcar/platform.mk]: have you tested this commit? it affects the boot flow (check bl2_el3_entrypoint.S). IIRC this was an issue

**Comment by antonio-nino-diaz-arm** [comment]
Hmm either I got confused with the conflicts of another PR or git could fix them by just rebasing, I don't know. In any case, this change is fine by me.

**Comment by antonio-nino-diaz-arm** [comment]
jenkins: test this please

**Comment by marex** [comment]
On 1/8/19 10:15 AM, Antonio Niño Díaz wrote:
> Trying to build FVP with clang with your changes fails:
> 
> |plat/arm/common/arm_bl2_setup.c:39:14: error: weak identifier
> 'bl2_plat_sec_mem_layout' never declared [-Werror] #pragma weak
> bl2_plat_sec_mem_layout ^ 1 error generated. Makefile:735: recipe for
> target 'build/fvp/debug/bl2/arm_bl2_setup.o' failed |
> 
> I think you'll have to drop this patch:
> 
> |rcar_gen3: plat: Function cleanup Replace foo_t with struct foo and
> drop unused function prototypes. |

Should be fixed now, the patch is only partly dropped. I added two clang
fixes too.

-- 
Best regards,
Marek Vasut

**Comment by antonio-nino-diaz-arm** [comment]
jenkins: test this please

**Comment by antonio-nino-diaz-arm** [comment]
The failures are just static checks because of modifying the staging drivers.

**Pull request merged**